### PR TITLE
change from built-in webpack uglify to uglifyJS webpack plugin

### PR DESCRIPTION
**Pull request by cezaraugusto**
this was making builds impossible due to uglify not properly transpiling ES6

fix https://github.com/brave/brave-browser/issues/309

test plan:

```
yarn build Release --debug_build=false --official_build=true --channel=beta
```

also as a double check ensure new tab page shows no console errors

### CODE REVIEW

**Comment by cezaraugusto** [comment]
security audit asked in https://github.com/brave/internal/issues/367

File: components/webpack/prod.config.js [hunk 1]
```diff
@@ -33,7 +39,10 @@ module.exports = {
     rules: [
       {
         test: /\.tsx?$/,
-        loader: 'awesome-typescript-loader'
+        loader: 'awesome-typescript-loader',
+        query: {
+          useBabel: true
```
Comment (bbondy) [inline components/webpack/prod.config.js]: do we need babel here?

Comment (bbondy) [reply components/webpack/prod.config.js]: If not pls PR to remove, but I'll merge this to unblock devops. Thanks.

Comment (cezaraugusto) [reply components/webpack/prod.config.js]: added https://github.com/brave/brave-core/pull/152

**Pull request merged**